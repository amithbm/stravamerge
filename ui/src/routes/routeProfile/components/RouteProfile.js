import React from 'react'
import { Grid, Row, Col } from 'react-bootstrap'
import RouteProfileRendererHost from './RouteProfileRendererHost'
import ActivityListItem from '../../activities/components/ActivityListItem'

const RouteProfile = ({activity}) => <Grid>
    <Row>
        <Col xs={12}>
            {activity ?
                <div>
                    <ActivityListItem activity={activity} includeHyperlinks={false} />
                        <RouteProfileRendererHost activity={activity} />
                    </div> : <div></div>
            }
        </Col>
    </Row>
</Grid>

RouteProfile.propTypes = {
    activity: React.PropTypes.object
}

export default RouteProfile
