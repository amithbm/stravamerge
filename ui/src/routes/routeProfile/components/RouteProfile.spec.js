import React from 'react'
import {shallow} from 'enzyme'
import RouteProfile from './RouteProfile'
import toJson from 'enzyme-to-json'

describe('components', () => {
    describe('<RouteProfile />', () => {
        it('renders content when activity set', () => {
            const component =  shallow(<RouteProfile activity={{ activityId: 1 }} />)
            const tree = toJson(component)
            expect(tree).toMatchSnapshot()
        })

        it('renders empty when no activity', () => {
            const component =  shallow(<RouteProfile />)
            const tree = toJson(component)
            expect(tree).toMatchSnapshot()
        })
    })
})

