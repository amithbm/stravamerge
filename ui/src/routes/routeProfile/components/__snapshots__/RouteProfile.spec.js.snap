exports[`components <RouteProfile /> renders content when activity set 1`] = `
<Grid
  bsClass="container"
  componentClass="div"
  fluid={false}>
  <Row
    bsClass="row"
    componentClass="div">
    <Col
      bsClass="col"
      componentClass="div"
      xs={12}>
      <div>
        <Component
          activity={
            Object {
              "activityId": 1,
            }
          }
          includeHyperlinks={false} />
        <RouteProfileRendererHost
          activity={
            Object {
              "activityId": 1,
            }
          } />
      </div>
    </Col>
  </Row>
</Grid>
`;

exports[`components <RouteProfile /> renders empty when no activity 1`] = `
<Grid
  bsClass="container"
  componentClass="div"
  fluid={false}>
  <Row
    bsClass="row"
    componentClass="div">
    <Col
      bsClass="col"
      componentClass="div"
      xs={12}>
      <div />
    </Col>
  </Row>
</Grid>
`;
